tests/st/calicoctl/test_crud.py

```diff
@@ -557,6 +557,47 @@ def test_clusterinfo(self):
         rc = calicoctl("delete clusterinfo %s" % name(clusterinfo_name1_rev1))
         rc.assert_error(NOT_SUPPORTED)
 
+    def test_metadata_unchanged(self):
+        """
+        Test that the metadata fields other than labels and annotations cannot be changed
+        in create and update operations.
+        """
+        # Use apply to create a new Host Endpoint and get it to determine the
+        # current resource version (first checking that it doesn't exist).
+        rc = calicoctl(
+            "get hostendpoint %s -o yaml" % name(hostendpoint_name1_rev2))
+        rc.assert_error(text=NOT_FOUND)
+
+        rc = calicoctl("apply", data=hostendpoint_name1_rev2)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get hostendpoint %s -o yaml" % name(hostendpoint_name1_rev2))
+        rc.assert_no_error()
+        rev0 = rc.decoded
+
+        # Apply the Host Endpoint (with no resource version) and get it to
+        # assert the resource version is not the same.
+        rc = calicoctl("apply", data=hostendpoint_name1_rev3)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get hostendpoint %s -o yaml" % name(hostendpoint_name1_rev3))
+        rc.assert_no_error()
+        rev1 = rc.decoded
+        self.assertNotEqual(rev0['metadata']['resourceVersion'], rev1['metadata']['resourceVersion'])
+
+        # Validate that only annotations and labels were changed in the metadata
+        self.assertNotEqual(rev1['metadata'].get('selfLink', ''), hostendpoint_name1_rev3['metadata'].get('selfLink', ''))
+        self.assertNotEqual(rev1['metadata'].get('uid', ''), hostendpoint_name1_rev3['metadata'].get('uid', ''))
+        self.assertNotEqual(rev1['metadata'].get('generation', ''), hostendpoint_name1_rev3['metadata'].get('generation', ''))
```

Comment:
I don't see the point in comparing to the original data since we know calicoctl is not going to set any of the metadata fields from that data except the annotations and labels.

Reply:
The thing I was trying to prove here was that they weren't set so that they wouldn't match the original but that's a little hokey.  I think I'll change the assertion to just say that the keys don't exist in the `metadata`